extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -137,11 +146,34 @@ export class AmpUserNotification extends AMP.BaseElement {
 
     this.storageKey_ = 'amp-user-notification:' + this.elementId_;
 
+    this.showIfGeo_ = this.element.getAttribute('data-show-if-geo');
+    this.showIfNotGeo_ = this.element.getAttribute('data-show-if-not-geo');
+
     this.showIfHref_ = this.element.getAttribute('data-show-if-href');
     if (this.showIfHref_) {
       assertHttpsUrl(this.showIfHref_, this.element);
     }
 
+    // Casts string to boolean using !!(string) then coerce that to
+    // number using when we add them so we can see easily test
+    // how many flags were set.  We want 0 or 1.
+    user().assert(
+        !!this.showIfHref_ +
+        !!this.showIfGeo_ +
+        !!this.showIfNotGeo_ <= 1,
+        'Only one "data-show-if-*" attribute allowed'
+    );
+
+    if (this.showIfGeo_) {
```

Comment:
A comment here. It is difficult for a publisher to notice misconfiguration or typo because we never check if the geoGroup specified with `data-show-if-geo` is actually defined within `<amp-geo>`
Are we planning to have the geoService provides a list of user defined geoGroups? Or we think it's their duty to make sure geoGroups specified in amp-user-notifiation/amp-consent are properly defined.

Reply:
I think that would be a good addition to amp-geo - I'll open a separate issue for it.

Reply:
#15705